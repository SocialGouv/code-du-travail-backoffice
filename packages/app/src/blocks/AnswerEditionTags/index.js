import styled from "@emotion/styled";
import React from "react";
import { Flex } from "rebass";

import _Subtitle from "../../elements/Subtitle";
import Tag from "../../elements/Tag";

const Container = styled(Flex)`
  color: #555555;
  flex-grow: 1;
  padding: 1rem;
`;
const Subtitle = styled(_Subtitle)`
  font-size: 0.875rem;
  font-weight: 700;
`;

export default class AnswerEditionTagsBlock extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      selectedTags: this.props.selectedTags
    };
  }

  toggleTag(id) {
    if (this.state.selectedTags.includes(id)) {
      this.setState({
        selectedTags: this.state.selectedTags.filter(_id => _id !== id)
      });
      this.props.onToggle(id, false);

      return;
    }

    this.setState({ selectedTags: [...this.state.selectedTags, id] });
    this.props.onToggle(id, true);
  }

  getTags(category) {
    return this.props.tags
      .filter(tag => tag.category === category)
      .map(({ id, value }, index) => (
        <Tag
          key={index}
          onClick={() => this.toggleTag(id)}
          selected={this.state.selectedTags.includes(id)}
        >
          {value}
        </Tag>
      ));
  }

  render() {
    return (
      <Container className={this.props.className} flexDirection="column">
        <_Subtitle isFirst>Étiquettes</_Subtitle>
        <p>
          Sélectionnez les étiquettes correspondant à la réponse que vous rédigez afin de nous aider
          à classifier le contenu:
        </p>

        <Flex flexDirection="column">
          <Subtitle isFirst>Type de contrat</Subtitle>
          <Flex flexWrap="wrap">{this.getTags("contract_type")}</Flex>
        </Flex>

        <Flex flexDirection="column">
          <Subtitle>Cible</Subtitle>
          <Flex flexWrap="wrap">{this.getTags("target")}</Flex>
        </Flex>

        <Flex flexDirection="column">
          <Subtitle>Durée de travail</Subtitle>
          <Flex flexWrap="wrap">{this.getTags("work_time")}</Flex>
        </Flex>

        <Flex flexDirection="column">
          <Subtitle>{"Type d'horaires"}</Subtitle>
          <Flex flexWrap="wrap">{this.getTags("work_schedule_type")}</Flex>
        </Flex>

        <Flex flexDirection="column">
          <Subtitle>Particularismes</Subtitle>
          <Flex flexWrap="wrap">{this.getTags("distinctive_identity")}</Flex>
        </Flex>
      </Container>
    );
  }
}
