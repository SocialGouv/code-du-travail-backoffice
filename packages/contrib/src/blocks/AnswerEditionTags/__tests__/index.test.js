import React from "react";
import { fireEvent, render } from "@testing-library/react";

import AnswerEditionTags from "..";

describe("[Contrib] blocks/<AnswerEditionTags />", () => {
  const props = {
    onToggle: jest.fn(),
    selectedTags: ["da68fa2d-4bfc-4b5b-bea3-f39f44719650"],
    tags: [
      {
        id: "da68fa2d-4bfc-4b5b-bea3-f39f44719650",
        value: "A Tag",
        category: "contract_type"
      },
      {
        id: "d7d54d79-b389-416c-9e06-43eb36a5ab30",
        value: "Another Tag",
        category: "target"
      }
    ]
  };

  const { container, getByText } = render(<AnswerEditionTags {...props} />);

  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });

  it("should trigger onToggle() with the expected param (on)", () => {
    fireEvent.click(getByText("Another Tag"));

    expect(props.onToggle).toHaveBeenCalledWith(props.tags[1].id, true);
  });

  it("should trigger onToggle() with the expected param (off)", () => {
    fireEvent.click(getByText("Another Tag"));

    expect(props.onToggle).toHaveBeenCalledWith(props.tags[1].id, false);
  });
});
