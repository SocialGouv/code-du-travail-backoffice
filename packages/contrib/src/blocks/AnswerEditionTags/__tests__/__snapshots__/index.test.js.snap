// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] blocks/<AnswerEditionTags /> should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH AnswerEditionTags__Container-ycvsa3-0 kGjiFp"
  >
    <h2
      class="Subtitle-sc-17ig8qv-0 gsnyDG"
    >
      Étiquettes
    </h2>
    <p>
      Sélectionnez les étiquettes correspondant à la réponse que vous rédigez afin de nous aider à classifier le contenu:
    </p>
    <div
      class="sc-bdVaJa sc-bwzfXH eVPJMi"
    >
      <h2
        class="Subtitle-sc-17ig8qv-0 AnswerEditionTags__Subtitle-ycvsa3-1 cVqAlR"
      >
        Type de contrat
      </h2>
      <div
        class="sc-bdVaJa sc-bwzfXH hbZeap"
      >
        <span
          class="Tag-sc-16selk6-0 lnmabc"
        >
          A Tag
        </span>
      </div>
    </div>
    <div
      class="sc-bdVaJa sc-bwzfXH eVPJMi"
    >
      <h2
        class="Subtitle-sc-17ig8qv-0 AnswerEditionTags__Subtitle-ycvsa3-1 eQbvOl"
      >
        Cible
      </h2>
      <div
        class="sc-bdVaJa sc-bwzfXH hbZeap"
      >
        <span
          class="Tag-sc-16selk6-0 fPHNXI"
        >
          Another Tag
        </span>
      </div>
    </div>
    <div
      class="sc-bdVaJa sc-bwzfXH eVPJMi"
    >
      <h2
        class="Subtitle-sc-17ig8qv-0 AnswerEditionTags__Subtitle-ycvsa3-1 eQbvOl"
      >
        Durée de travail
      </h2>
      <div
        class="sc-bdVaJa sc-bwzfXH hbZeap"
      />
    </div>
    <div
      class="sc-bdVaJa sc-bwzfXH eVPJMi"
    >
      <h2
        class="Subtitle-sc-17ig8qv-0 AnswerEditionTags__Subtitle-ycvsa3-1 eQbvOl"
      >
        Type d'horaires
      </h2>
      <div
        class="sc-bdVaJa sc-bwzfXH hbZeap"
      />
    </div>
    <div
      class="sc-bdVaJa sc-bwzfXH eVPJMi"
    >
      <h2
        class="Subtitle-sc-17ig8qv-0 AnswerEditionTags__Subtitle-ycvsa3-1 eQbvOl"
      >
        Particularismes
      </h2>
      <div
        class="sc-bdVaJa sc-bwzfXH hbZeap"
      />
    </div>
  </div>
</div>
`;
